lib/instream/connection/query_runner.ex

```diff
@@ -12,13 +17,33 @@ defmodule Instream.Connection.QueryRunner do
   Executes `:ping` queries.
   """
   @spec ping(Query.t, Keyword.t, Keyword.t) :: :pong | :error
-  def ping(%Query{} = query, _opts, conn) do
+  def ping(%Query{} = query, opts, conn) do
     headers = conn |> Headers.assemble()
 
-    conn
-    |> URL.ping(query.opts[:host])
-    |> :hackney.head(headers, "", Keyword.get(conn, :http_opts, []))
-    |> Response.parse_ping()
+    { query_time, response } = :timer.tc fn ->
+      conn
+      |> URL.ping(query.opts[:host])
+      |> :hackney.head(headers, "", Keyword.get(conn, :http_opts, []))
+    end
+
+    result = response |> Response.parse_ping()
+    status = case response do
+      { :ok, status, _ } -> status
+      _                  -> 0
```

Comment:
I think the exact return value does not really matter for `ping`/`status` calls. Right now only `write` and `query` will raise if the remote server is not available, otherwise it gets "silently" handled.

So the status `0` can only happen if the server is unavailable. Thinking about misconfigured hostnames oder unstarted services here. These events may be interesting to log in a separate way...